formatter/src/lib.rs

```diff
@@ -18,3 +18,8 @@ pub fn format_file(path: &Path, settings: FormatterSettings) -> Result<String, F
     let file = std::fs::read_to_string(path)?;
     format_file_source(&file, settings)
 }
+
+pub fn format_stdin(content: &[u8], settings: FormatterSettings) -> Result<String, FormatError> {
```

Comment:
This is basically a "format utf8 encoded bytes" method, as this formatter is not aware / does not care if it comes from stdin or not.

I think I prefer to export `format_file_source` (`pub use format_file_source` in lib.rs) and then calling that from the CLI crate, agree?